### DISCUSSION TITLE
Show Hidden Fields in GraphQL

### DISCUSSION THREAD

**Discussion by martinherweg** [solspace/craft-freeform#268] [Questions & Answers] [answered]
Hey,

i want to use a hidden Field in a Headless environment, unfortunately I can not get it into the request, even if I use the `... on FreeformField_Hidden` fragment.

Is there a setting I have overlooked to enable hidden fields in Graph QL?

Cheers,
Martin

**Comment by martinherweg** [comment, marked as answer]
Got it myself, I was using a `form > pages > rows > fields` query but found out hidden fields are always in the top level and need to queries seperately with a `form > fields` query.

While I think that is quite unintuitive for a Graph QL it answers my question.

**Reply by kjmartens** [reply]
Thanks for your feedback. We'll make a note about this and see if we can improve upon it in the future. 🙂
